Shop/Shop/Controllers/ProductController.cs

```diff
@@ -61,5 +83,26 @@ public Product Create(Product product)
 
             return resultProduct;
         }
+
+        private bool IsCorrectQuantity(Product product)
```

Comment:
it is class Controller. methods bellow not related to controller actions. move it to somewhere. look at SOLID. this case is first letter S.